spec/intl.html

```diff
@@ -419,7 +418,9 @@ <h1>FormatDateTimePattern ( _dateTimeFormat_, <ins>_pattern_</ins>, _patternPart
           1. Perform ! CreateDataPropertyOrThrow(_nf3Options_, *"minimumIntegerDigits"*, _fractionalSecondDigits_).
           1. Perform ! CreateDataPropertyOrThrow(_nf3Options_, *"useGrouping"*, *false*).
           1. Let _nf3_ be ? Construct(%NumberFormat%, &laquo; _locale_, _nf3Options_ &raquo;).
-        1. Let _tm_ be ToLocalTime(<del>_x_</del><ins>_epochNanoseconds_</ins>, _dateTimeFormat_.[[Calendar]], _dateTimeFormat_.[[TimeZone]]).
+        1. <ins>If _dateTimeFormat_.[[TimeZone]] is not *undefined*, then</ins>
+          1. <ins>Throw a *RangeError* exception.</ins>
```

Comment:
To implement option 8, I think the logic we want here is:
- If both _dateTimeFormat_.[[TimeZone]] and _timeZone_ are not undefined, throw a RangeError exception.
- If _timeZone_ is undefined, set _timeZone_ to _dateTimeFormat_.[[TimeZone]].
- If _timeZone_ is undefined, set _timeZone_ to DefaultTimeZone().

That means we also have to leave _dateTimeFormat_.[[TimeZone]] undefined if it's not explicitly specified in InitializeDateTimeFormat and not call DefaultTimeZone there. Probably in practice, implementations would still call DefaultTimeZone in InitializeDateTimeFormat, and keep track of whether the time zone was explicitly specified using a separate flag. We could choose to do that in the spec as well. It might look something like this:
- If _timeZone_ is not undefined and _dateTimeFormat_.[[TimeZoneSpecified]] is true, throw a RangeError exception.
- If _timeZone_ is undefined, set _timeZone_ to _dateTimeFormat_.[[TimeZone]].